### ISSUE TITLE
Save / Load Times in Single Player from Minecolonies (and Save Bloat)

### ISSUE THREAD

**Issue by WonderDog** [AllTheMods/ATM-3#485]
In ATM 3 v5.91 I have been having an issue with save load time getting longer the more I play in single player.

I have my instance running on a samsung 860 evo SSD which is not the system disk and after about weeks of casual play (day 100) it now takes over 5 minutes to save or load.  Longer in fact than loading minecraft.

This also happens on the Creative Flat World which has no explored regions (region files are 9.6MB compared to 160MB for my survival world).

I have not done anything with the Minecolonies mod and its folder in the save is nearly 300MB (It is nearby half the total size of my survival save). In my creative world it is nearly 75MB and is also over half my save size.

To fix the issues I remove the mod from the load list and save/load drops to about 30 seconds.  First load after removing the mod takes about a minute as it cleans up the missing mod stuff.
I then removed the minecolonies folder from the save.

In version 5.8 before I updated my world I had the same problem and removing fixed it also but on updating it was reinstalled.

**Comment by Boomflex** [comment]
In the 3 Worlds I have open the biggest folder is 2.08mb...

On the Official Public server it is 1.20mb and Official Patreon it is 2mb.

Not sure why this is happening and it is not reproducible.

Will keep an eye on this though.

**Comment by WonderDog** [comment]
I had a look at the mincolonies folder in my save game it has pages of date stamped zip files that is most of the space used.  (It was still in the Recycle bin)

They seem to be created every time that the game is loaded and seem to increase geometrically in size as time progress going from 1k to 8MB in my current game.  Might be something to do being single player.

I checked my archived backups but my script does not include zip files so I can't check the v5.8 save from before I deleted minecolonies.

Looking over my weekly archived backups the '_minecolonies.dat_' file in a new save is about 7kb, in my creative testing world in which I have not moved much from spawn I have 4mb, 14mb and 49mb in different archives while the region folder only has 4 dat files in all of them for about 4-5mb total.

I will test again when next version of ATM is released.  I will do a clean install as I have twitch end up with duplicate mod entries (referencing only one JAR) after an update.

**Comment by WonderDog** [comment]
Just did new install of ATM3 5.9.3 and still have the same issue

With a new world every time I start the world minecolonies creates new backup archive of _colonies.dat_.  This file grows fast starting at 400 bytes and reaching 16 kilobytes after two hours and three restarts of Minecraft.

Using NBT Explorer I notice that the problem is in **Compatibility Manager** / **Tag Saplings** which  grows geometricaly

In Quick test game where I have 3 backups I have the number of **TagSapling** entries : 0 (On World Creation), 339, 603, 1395.  Compare to **TagOre** 0, 84, 84, 84 and **TagSapLeaves** with 0, 0, 0, 0

I just downloaded another instance of ATM3 v 5.9.3 without optifine and made no changes to any settings / config. 

I created a new default world and exited after a minute and checked _colonies.dat_ in NBT explorer.  Did that three times and the number of entries of **TagSapling** 339, 867, 1923
